src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -282,13 +282,15 @@ internal static ExtensionBundleOptions GetExtensionBundleOptions(HostBuilderCont
             return options;
         }
 
-        private static void RegisterFileProvisioningService(IHostBuilder builder)
+        private static void RegisterFileProvisioningService(IHostBuilder builder, ILoggerFactory loggerFactory)
         {
             if (string.Equals(Environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName), "powershell"))
             {
+                var logger = loggerFactory.CreateLogger<FuncAppFileProvisionerFactory>() ?? throw new ArgumentNullException(nameof(loggerFactory));
+                var funcAppFileProvisionerFactory = new FuncAppFileProvisionerFactory(logger);
```

Comment:
Do not create logger here. ILoggerFactory should be part of the constructor for FuncAppFileProvisionerFactory

Reply:
Fixed.